samples: philosophers: Enable thread awareness (#38087)
zephyrproject-rtos/zephyr: samples/philosophers/README.rst

   Philosopher 5 [C:-2]  THINKING [ 1700 ms ]

Exit QEMU by pressing :kbd:`CTRL+A` :kbd:`x`.

> Debug Threads
> ********************

Reviewer: ```suggestion
Debug Threads
*************
```

> The Philosopher sample by default enable CONFIG_DEBUG_THREAD_INFO.  This allows

Reviewer 2: ```suggestion
The philosophers sample by default enables CONFIG_DEBUG_THREAD_INFO. This allows
```

Reviewer: ```suggestion
The Philosopher sample by default enable :kconfig:`CONFIG_DEBUG_THREAD_INFO`.  This allows
```
